### PR TITLE
payments support added

### PR DESCRIPTION
**Pull request by nasimafeef**
added functions for payments
payments_ready variable set on payments ready

### CODE REVIEW

**Review by vivekb91 — APPROVED** [review]

File: js/index.js [hunk 1]
```diff
@@ -11,6 +11,12 @@ var FacebookInstant = Class(function () {
         FBInstant.setLoadingProgress(100);
         FBInstant.startGameAsync()
         .then(function (){
+          this.FBInstant.payments.onReady(function (){
+            if (this.FBInstant.getSupportedAPIs()
+              .includes('payments.purchaseAsync')) {
+              GC.app.payments_ready = true;
```
Comment (vivekb91) [inline js/index.js]: can we move this to callback

File: js/index.js [hunk 2]
```diff
@@ -11,6 +11,12 @@ var FacebookInstant = Class(function () {
         FBInstant.setLoadingProgress(100);
         FBInstant.startGameAsync()
         .then(function (){
```
Comment (rampr) [inline js/index.js]: indentation also require a space before `{`

File: js/index.js [hunk 3]
```diff
@@ -11,6 +11,12 @@ var FacebookInstant = Class(function () {
...
+          this.FBInstant.payments.onReady(function (){
+            if (this.FBInstant.getSupportedAPIs()
+              .includes('payments.purchaseAsync')) {
+              opts.payments_ready = true;
+            }
+          });
           cb(opts);
         })
       });
```
Comment (rampr) [inline js/index.js]: This chaining does not seem correct.

Comment (rampr) [reply js/index.js]: Something like this: 

```
this.initialise = function (cb, opts) {
  var FBInstant = this.FBInstant;

  FBInstant.initializeAsync()
    .then(function () {
        FBInstant.setLoadingProgress(100);
        return FBInstant.startGameAsync();
     })
     .then(function () {
       this.FBInstant.payments.onReady(function () {
         if (this.FBInstant.getSupportedAPIs()
                .includes('payments.purchaseAsync')) {
           opts.payments_ready = true;
         }
        });
        cb(opts);
      })
};
```

File: js/index.js [hunk 4]
```diff
@@ -126,10 +142,10 @@ var FacebookInstant = Class(function () {
   this.getRewardedVideoAsync = this.FBInstant.getRewardedVideoAsync;
 
   // Returns a promise that resolves with whether the player can subscribe to the game bot or not.
-  this.canSubscribeBotAsync = this.FBInstant.canSubscribeBotAsync;
```
Comment (rampr) [inline js/index.js:129]: is this breaking the live game ?

Comment (nasimafeef) [reply js/index.js:129]: nop. yesterday vivek fixed these and pushed changes. still it's showing in this comparison.

File: js/index.js [hunk 5]
```diff
@@ -3,17 +3,20 @@ var FacebookInstant = Class(function () {
...
-        FBInstant.setLoadingProgress(100);
-        FBInstant.startGameAsync()
-        .then(function (){
-          cb(opts);
-        })
-      });
+        return FBInstant.setLoadingProgress(100);
+      })
+      .then(FBInstant.startGameAsync)
+      .then(bind(this, function (){
```
Comment (rampr) [inline js/index.js]: space before {

**Review by rampr — APPROVED** [review]

**Pull request merged**